### PR TITLE
Add elixir package to Mariner to support rabbitmq

### PR DESCRIPTION
**Pull request by sameluch**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Add the elixir package to Mariner in preparation for adding rabbitmq

###### Change Log  

Add elixir package to Mariner, add spec file and update license files accordingly.

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- AMD64 Buddy Build ID: 323199
- ARM64 Buddy Build ID: 323200

### CODE REVIEW

File: SPECS/elixir/elixir.spec [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+%description
+elixir programming language
+
+%prep
+%setup -q -n elixir-%{version}
+
+%build
+export LANG="en_US.UTF-8"
+make
```
Comment (mfrw) [inline SPECS/elixir/elixir.spec:23]: How about ?
```suggestion
%make_build
```

File: SPECS/elixir/elixir.spec [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+URL:            https://elixir-lang.org
+Source0:        https://github.com/elixir-lang/elixir/archive/v%{version}/elixir-%{version}.tar.gz
+BuildRequires:  erlang
+BuildRequires:  glibc-lang
+
+%description
+elixir programming language
+
+%prep
+%setup -q -n elixir-%{version}
```
Comment (mfrw) [inline SPECS/elixir/elixir.spec:19]: we could probably drop the `-n elixir-%{version}` as it is the default :)
```suggestion
%autosetup
```

**Review by mfrw — APPROVED** [review]
LGTM, a few suggestions though.

File: SPECS/elixir/elixir.spec [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+%{_libdir}/elixir/*
+%{_mandir}/man1/elixir.1.gz
+%{_mandir}/man1/elixirc.1.gz
+%{_mandir}/man1/iex.1.gz
+%{_mandir}/man1/mix.1.gz
+
+
+%changelog
+* Mon Feb 27 2023 Sam Meluch <sammeluch@microsoft.com> - 1.14.3-1
+- Original version for CBL-Mariner
```
Comment (mfrw) [inline SPECS/elixir/elixir.spec:43]:
```suggestion
- Original version for CBL-Mariner
- License verified
```

File: SPECS/elixir/elixir.spec (hunk 1 above)
Comment (sameluch) [reply SPECS/elixir/elixir.spec]: Updated to use the macro

File: SPECS/elixir/elixir.spec (hunk 2 above)
Comment (sameluch) [reply SPECS/elixir/elixir.spec]: Updated to use the autosetup macro

File: SPECS/elixir/elixir.spec (hunk 3 above)
Comment (sameluch) [reply SPECS/elixir/elixir.spec:43]: Updated changelog comment, and ensured the license is verified

**Pull request merged**